source/server/listener_impl.h

```diff
@@ -224,13 +225,15 @@ class ListenerImpl : public Network::ListenerConfig,
   const uint64_t hash_;
   ProtobufMessage::ValidationVisitor& validation_visitor_;
 
+  // This init watcher, if available, notifies the "parent" listener manager when listener
+  // initialization is complete. It may be reset to cancel interest.
+  std::unique_ptr<Init::WatcherImpl> init_watcher_;
+  // A target is added to Server's InitManager if workers_started_ is false.
+  std::unique_ptr<Init::TargetImpl> listener_init_target_;
```

Comment:
This doesn't really need to be a `unique_ptr` either. I'd recommend always creating the target, so that you don't have to worry about `ASSERT(listener_init_target_ != nullptr)`, and just add it to the parent's server's init manager if workers haven't been started.